### PR TITLE
Add: shipping plugins in shipping task based on location

### LINKED ISSUE #36248: Add Skydropx, Envia, Sendcloud, Packlink to shipping task
P2: p73Jol-1C5-p2#comment-6601
Figma: cJQeyyhx9PraQrY3YAtCGM-fi-4001%3A345877

Incorporate new shipping partners into the shipping task. The partners will be displayed dynamically based on the store's country, and the design will be modified depending on whether one or two partners are featured.

Example 2-partners: France & Germany, Sendcloud & Packlink

Image

Example 1-partner (existing ShipStation): Packlink

Image

## Acceptance Criteria

1. Dynamically feature partners for countries based on the provided list (p73Jol-1C5-p2#comment-6601)
1. When featuring a single partner, add their logo to the existing shipping task cJQeyyhx9PraQrY3YAtCGM-fi-4001%3A345877
1. When featuring two partners, use a new design cJQeyyhx9PraQrY3YAtCGM-fi-4001%3A345877
1. Currently Envia and Skydropx plugins are not available on wp.org. Replace the CTAs with opening up URLs instead and copy is `Download`

## Other details

Following are the countries with 2 partners displayed. Left-right ordering is important (p73Jol-1C5-p2#comment-6695):

- France & Germany: Sendcloud & Packlink
- Spain & Italy: Packlink & Sendcloud
- UK: Sendcloud & Shipstation

Plugin URLs:

1. Sendcloud: https://wordpress.org/plugins/sendcloud-shipping/
3. Packlink: https://wordpress.org/plugins/packlink-pro-shipping/
4. Envia: **Opens in new URL** https://woocommerce.com/products/envia-shipping-and-fulfillment/
5. Skydropx: https://wordpress.org/plugins/skydropx-cotizador-y-envios/

### PR DESCRIPTION
**Pull request by rjchow**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

See https://github.com/woocommerce/woocommerce/issues/36248 for full background.

This PR adds the functionality to feature more shipping partners in the onboarding shipping task.

Closes #36248 .

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

- [ ] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

1. Set up a new WooCommerce installation
2. During the onboarding wizard, select one of the countries in the list for testing, with a physical product and without additional plugins 
3. On the WooCommerce admin home screen, look for the "Add shipping costs" task and click on it
4. It should bring you to the shipping task page where you can add a shipping partner
5. The third step of the stepper should be a panel with either one or two shipping providers depending on the country you selected earlier
6. The buttons on the panels should successfully install and activate the plugin, except for Envia which has a manual installation method.
7.  A successful installation should occur and bring you back to the homescreen.

Note: The Skydropx provider has an installation issue right now which fails on the first try and succeeds on retry. This will be resolved by either the plugin developers fixing it before this is merged, or otherwise we will revert to the manual installation instructions which does not automatically install the plugin.

https://user-images.githubusercontent.com/27843274/220021752-7613b6e1-c8ae-4071-be03-b80e9cc30158.mp4

### Countries

This is the list of countries and which partners will be featured.
LATAM

    Mexico: Skydropx
    Colombia: Skydropx
    Chile: Envia
    Argentina: Envia
    Peru: Envia
    Brazil: Envia
    Uruguay: Envia
    Guatemala: Envia

Note that in the future, as Skydropx expands geographically, we would also add them to countries where Envia is being promoted. Both would be featured side by side.
Europe

    France: Sendcloud & Packlink
    Spain: Sendcloud & Packlink
    Italy: Sendcloud & Packlink
    Germany: Sendcloud & Packlink
    UK: Sendcloud & ShipStation
    Netherlands: Sendcloud
    Austria: Sendcloud
    Belgium: Sendcloud

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Shipping.php [hunk 1]
```diff
@@ -120,7 +120,7 @@ public function can_view() {
 				return true;
 			}
 
-			return in_array( $store_country, array( 'AU', 'CA', 'GB' ), true );
+			return in_array( $store_country, array( 'US', 'CA', 'AU', 'GB', 'ES', 'IT', 'DE', 'FR', 'MX', 'CO', 'CL', 'AR', 'PE', 'BR', 'UY', 'GT', 'NL', 'AT', 'BE' ), true );
```
Comment (chihsuan) [inline plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Shipping.php]: Is there any reason that we need to add the US to the list here? I think we apply the smart default for the US.

File: packages/js/components/src/plugins/index.tsx [hunk 2]
```diff
@@ -25,6 +25,11 @@ type PluginsProps = {
 	pluginSlugs?: string[];
 	onAbort?: () => void;
 	abortText?: string;
+	installText?: string;
+	installButtonVariant?: 'primary' | 'secondary' | 'tertiary' | 'link';
+	learnMore?: string;
```
Comment (chihsuan) [inline packages/js/components/src/plugins/index.tsx]: How about renaming this to `learnMoreLink`? I was confused when I first saw it and learnMoreText.

File: packages/js/components/src/plugins/index.tsx [hunk 3]
```diff
@@ -25,6 +25,11 @@ type PluginsProps = {
 	pluginSlugs?: string[];
 	onAbort?: () => void;
 	abortText?: string;
+	installText?: string;
+	installButtonVariant?: 'primary' | 'secondary' | 'tertiary' | 'link';
```
Comment (chihsuan) [inline packages/js/components/src/plugins/index.tsx]:
```suggestion
	installButtonVariant?: Button.BaseProps[ 'variant' ];
```

File: plugins/woocommerce-admin/client/tasks/fills/shipping/index.js [hunk 4]
```diff
@@ -397,71 +421,146 @@ export class Shipping extends Component {
...
+												( element ) =>
+													element !== undefined
+											); // remove undefineds
+											// TODO: if pluginsForPartner is empty then we show a CTA with the URL instead
+											return pluginToPromote[
+												'dual-partner-layout'
+											]( {
+												children: (
+													<div className="woocommerce-task-shipping-recommendations_plugins-buttons">
+														<Plugins
```
Comment (chihsuan) [inline plugins/woocommerce-admin/client/tasks/fills/shipping/index.js:458]: 708 lines in this file with these nested blocks. 😱 We should refactor it in a follow-up task.

**Review by chihsuan — COMMENTED** [review]
Thanks for working on this! 👍  @rjchow @moon0326 

I just reviewed the code, haven't tested it.

**Comment by chihsuan** [comment]
> During the onboarding wizard, select one of the countries in the list for testing, with a physical product and without additional plugins
> The third step of the stepper should be a panel with either one or two shipping providers depending on the country you selected earlier

Could we update the test instructions with more details? For example, add the testing county list and expected shipping providers info. Otherwise, I don't think globalstep can test it.

File: plugins/woocommerce-admin/client/tasks/fills/shipping/index.js (hunk 4 above)
Comment (rjchow) [reply plugins/woocommerce-admin/client/tasks/fills/shipping/index.js:458]: Completely agree with that 🥹
It's a bit of a spaghetti ball right now and I think some more changes will be needed when we change it to use a remote API instead

File: packages/js/components/src/plugins/index.tsx (hunk 3 above)
Comment (rjchow) [reply packages/js/components/src/plugins/index.tsx]: Nice catch! Thanks!

File: packages/js/components/src/plugins/index.tsx (hunk 2 above)
Comment (rjchow) [reply packages/js/components/src/plugins/index.tsx]: Will do!

File: plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Shipping.php (hunk 1 above)
Comment (rjchow) [reply plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Shipping.php]: I think I got mixed up! Thanks for catching this

**Comment by rjchow** [comment]
> > During the onboarding wizard, select one of the countries in the list for testing, with a physical product and without additional plugins
> > The third step of the stepper should be a panel with either one or two shipping providers depending on the country you selected earlier
> 
> Could we update the test instructions with more details? For example, add the testing county list and expected shipping providers info. Otherwise, I don't think globalstep can test it.

Ah ok I'll move it down from the description then

**Review by adrianduffell — DISMISSED** [review]
Nice work on this @rjchow and @moon0326!  It tested well for me using Mexico, Australia, and Spain countries 🚀

**Review by chihsuan — DISMISSED** [review]
Tested well for me too! 🚀

**Review by adrianduffell — APPROVED** [review]
The latest change (Skydropx as manual install) LGTM!  🚀

**Pull request merged**